clinical-domain-agent/src/e2e/java/care/smith/fts/cda/AbstractCohortSelectorE2E.java

```diff
@@ -0,0 +1,456 @@
+    tca.start();
+    rda.start();
+    cda.start();
+
+    configureCdHdsMocks();
+    configureTcaMocks();
+    configureRdaMocks();
+  }
+
+  /**
+   * Configure RDA mocks. Default implementation should work for most cases. Override if specific
+   * RDA mock behavior is needed.
+   */
+  protected void configureRdaMocks() {
+    var rdaWireMock = new WireMock(rda.getHost(), rda.getPort());
+    rdaWireMock.register(
+        post(urlEqualTo("/api/v2/process/example/patient"))
+            .willReturn(
+                aResponse()
+                    .withStatus(202)
+                    .withHeader(CONTENT_TYPE, "application/fhir+json")
+                    .withHeader(
+                        CONTENT_LOCATION,
+                        "http://rd-agent:8080/api/v2/process/status/transfer-123")));
+
+    rdaWireMock.register(
+        get(urlEqualTo("/api/v2/process/status/transfer-123"))
+            .willReturn(aResponse().withStatus(200)));
+  }
+
+  @AfterEach
+  void tearDown() {
+    resetWireMockMappings();
+    stopContainers();
+  }
+
+  protected void resetWireMockMappings() {
+    if (cdHds.isRunning()) {
+      var cdHdsWireMock = new WireMock(cdHds.getHost(), cdHds.getPort());
+      cdHdsWireMock.resetMappings();
+    }
+    if (tca.isRunning()) {
+      var tcaWireMock = new WireMock(tca.getHost(), tca.getPort());
+      tcaWireMock.resetMappings();
+    }
+    if (rda.isRunning()) {
+      var rdaWireMock = new WireMock(rda.getHost(), rda.getPort());
+      rdaWireMock.resetMappings();
+    }
+  }
+
+  protected void stopContainers() {
+    // Stop containers in reverse order
+    if (cda.isRunning()) {
+      cda.stop();
+    }
+    if (rda.isRunning()) {
+      rda.stop();
+    }
+    if (tca.isRunning()) {
+      tca.stop();
+    }
+    if (cdHds.isRunning()) {
+      cdHds.stop();
+    }
+    network.close();
+  }
+
+  /**
+   * Creates status polling logic that polls the given content location URL until a final state is
+   * reached.
+   *
+   * @param webClient the WebClient to use for polling
+   * @param objectMapper the ObjectMapper to parse responses
+   * @param contentLocation the status URL to poll
+   * @return a Flux that emits status responses until completion
+   */
+  protected Flux<String> createStatusPoller(
+      WebClient webClient, ObjectMapper objectMapper, String contentLocation) {
+    return Flux.interval(Duration.ofSeconds(1))
+        .take(10) // Poll for maximum 10 seconds
+        .flatMap(tick -> performSingleStatusPoll(webClient, objectMapper, contentLocation, tick))
+        .takeUntil(statusResponse -> isTerminalPhase(objectMapper, statusResponse));
+  }
+
+  private reactor.core.publisher.Mono<String> performSingleStatusPoll(
+      WebClient webClient, ObjectMapper objectMapper, String contentLocation, long tick) {
+    log.info("Polling attempt: {}", tick + 1);
+    return webClient
+        .get()
+        .uri(contentLocation)
+        .retrieve()
+        .bodyToMono(String.class)
```

Comment:
I didn't see it earlier, but this is a `TransferProcessStatus`. We don't need to use `String` here and therefor we also don't need an ObjectMapper.